mod/metrics/lib/javascript/script_metrics.js.coffee

```diff
@@ -49,38 +72,62 @@ decko.editorContentFunctionMap['.wikiRating-editor'] = ->
 
 wikiRatingEditorHash = (table) ->
   hash = {}
-  table.find("tbody tr").each ->
+  variableMetricRows(table).each ->
     tr = $(this)
     if key = tr.find(".metric-label .thumbnail").data "cardName"
       hash[key] = tr.find(".metric-weight input").val()
-  hash
+  hash 
+
+# if all values are equals active the equalize
+activeEqualize = () -> 
+  if !valueFormEqual( $('.wikiRating-editor') )  
+    $('#equalizer').prop('checked', false)
+  else 
+    $('#equalizer').prop('checked', true)
+    toEqualize( $('.wikiRating-editor') )
+
+toEqualize = (table) -> 
+  val = (100 / (variableMetricRows(table).length - 1)).toFixed(2)
+    
+  variableMetricRows(table).each ->
+    tr = $(this)
+    tr.find('td.metric-weight').find('input').val(val)
+
+  validateWikiRating(table)
+
+variableMetricRows = (table) ->
+  table.find("tbody tr")
 
 $(window).ready ->
   $('body').on 'input', '.metric-weight input', (_event) ->
     validateWikiRating $(this).closest(".wikiRating-editor")
 
   $('body').on "click", "._remove-weight", () ->
     removeWeightRow $(this).closest("tr")
+    toEqualize(  $('.wikiRating-editor') )
+  
 
 validateWikiRating = (table) ->
   hash = wikiRatingEditorHash table
   valid = tallyWeights table, hash
+  #toEqualize table 
   updateWikiRatingSubmitButton table.closest('form.card-form'), valid
 
 
 DIGITS_AFTER_DECIMAL = 2
 
 tallyWeights = (tbody, hash) ->
```

Comment:
this function is a little long for me.  We don't yet have strict rules for CoffeeScript (CodeClimate enforces rules for Ruby), but I start getting uncomfortable when a function is more than 5 lines, and usually I prefer 3 or less.

I would suggest separating this into two functions:  inputsAreValid and totalIsValid.

Reply:
...and you should write them so that they can both be tested with unit tests.